plugins/modules/ssh_config.py

```diff
@@ -274,6 +283,15 @@ def ensure_state(self):
             controlpersist=fix_bool_str(self.params.get('controlpersist')),
             dynamicforward=self.params.get('dynamicforward'),
         )
+        if self.params.get('other_options'):
+            for key, value in self.params.get('other_options').items():
+                if key not in args:
+                    if not isinstance(value, string_types):
+                        self.module.fail_json(msg="The other_options value provided for key {key!r} must be a string, got {type}".format(key=key,
+                                                                                                                                         type=type(value)))
+                    args[key] = value
+                else:
+                    self.module.fail_json(msg="Multiple values provided for key {key!r}".format(key=key))
```

Comment:
One more thing I'm wondering about: what if a user provides `DynamicForward` (in the above dict its spelled `dynamicforward`)? Or a user provides both `serveralivecountmax` and `ServerAliveCountMax` in `other_options`? Right now the module won't see that they're basically the same to SSH, and serialize both of them (or maybe some other code in the module will fail). In any case, during idempotency checks this will likely cause problems.

Maybe we should normalize all keys to lower-case? (Then we have to keep track of the original spelling when telling the user about errors/collisions.)

Reply:
Yes, good point. Have pushed a new version which fails if a non lower case key value is provided, and states keys must be lower case only in docstring. Think that will address this well enough?